aten/src/ATen/detail/ScalarTypeConversions.h

```diff
@@ -9,7 +9,7 @@ namespace at { namespace detail {
 
 template <typename T>
 inline T load(const void* data, ScalarType src_type) {
-  return AT_DISPATCH_ALL_TYPES(src_type, "load", [&]() {
+  return AT_DISPATCH_ALL_TYPES_AND(at::ScalarType::Half, src_type, "load", [&]() {
```

Comment:
it doesn't look like this is used anywhere but `store` below should probably be able to work on Half type.